tsl/src/bgw_policy/job_api.c

```diff
@@ -235,6 +286,35 @@ job_alter(PG_FUNCTION_ARGS)
 	if (!PG_ARGISNULL(6))
 		job->fd.config = PG_GETARG_JSONB_P(6);
 
+	if (!PG_ARGISNULL(9))
+	{
+		if (OidIsValid(check))
+		{
+			check_name_str = get_func_name(check);
+			if (check_name_str == NULL)
+				ereport(ERROR,
+						(errcode(ERRCODE_UNDEFINED_OBJECT),
+						 errmsg("function with OID %d does not exist", check)));
+
+			if (pg_proc_aclcheck(check, GetUserId(), ACL_EXECUTE) != ACLCHECK_OK)
+				ereport(ERROR,
+						(errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
+						 errmsg("permission denied for function \"%s\"", check_name_str),
+						 errhint("Job owner must have EXECUTE privilege on the function.")));
+
+			namestrcpy(&check_schema, get_namespace_name(get_func_namespace(check)));
+			namestrcpy(&check_name, check_name_str);
+
+			namestrcpy(&job->fd.check_schema, NameStr(check_schema));
+			namestrcpy(&job->fd.check_name, NameStr(check_name));
+			snprintf(schema_qualified_check_name,
+					 sizeof schema_qualified_check_name / sizeof schema_qualified_check_name[0],
```

Comment:
Nit: for readability
```suggestion
					 sizeof(schema_qualified_check_name) / sizeof(schema_qualified_check_name[0]),
```